### DISCUSSION TITLE
Linux flathub support?

### DISCUSSION THREAD

**Discussion by yotamguttman** [friction2d/friction#255] [Q&A]
I'm very impressed with the programme and can't wait to try it! would we eventually be able to download it from flathub on Linux?

**Comment by rodlie** [comment]
I don't have any plans to support more package formats than we already have.

What are the advantages compared to the existing binaries?

*IMHO sandboxing create more issues than it solves.*

**Reply by yotamguttman** [reply]
I trust you on that, definitely your call.

generally the Linux community tends to favour flatpak for security reasons and ease of use. mainly from a user's perspective, downloading the app is as simple as searching the name in flathub, pressing install, and then all the updates you release are taken care of automatically by their OS. flatpak is the most popular software package on many distros and it offers offers flathub which is essentially a universal app store for Linux, so in terms of discoverability, it's great but mainly, it makes the user's life much simpler, especially if they're no experts in managing their programmes.

**Reply by yotamguttman** [reply]
p.s. 
for instance, when I ditched Windows and started out on Linux (fedora), the community members strongly advised me to avised me to avoid installing any programmes from any sources other than flathub because when your knowledge in managing your os is limited, it can cause problems or compromise the system security. and it's not that I'm a clueless user, I'd used a computer for about 2 decades by the time I moved to Linux and for this reference, Adobe after effects for maybe half that time so I knew what I was doing just had to pick up Linux. flatpak is reviewed for security but it's also insanely easy to install/uninstall and manage. so even users who aren't keen to learn how Linux work and just want to use it, would have the easiest time to get Friction2D of flathub other than anything else out there offered for Linux.

**Reply by rodlie** [reply]
I know how flatpak works, so auto updates are the only pro?

Let me begin with the cons:

1. Will break svg preview
2. Will break color picker
3. Access to input/output files are restricted
4. Access to plugin are restricted

To fix the now new issues I will need to implement support for portals etc, additional work I'm not willing to do.

I might seem negative, I'm not :smile:  It's just that supporting flatpak (or snap) will require a lot of extra work to get on-par with the other binaries we have.

**Reply by rodlie** [reply]
> it can cause problems or compromise the system security.

Almost every app on flathub needs access to your $HOME to be functional, at that point it's all just a false sense of security imho.

Friction is a self-contained application without network access, don't see any big security issue here :)

https://friction.graphics/privacy.html

But, I understand your points, they are fair.

**Reply by yotamguttman** [reply]
please don't get me wrong. I wasn't saying that your app is a security threat if this is how it came out. 
as I said before I trust you that you know what's best for the project in the time being. maybe one day, when it's ready for an official release, you could reconsider it. otherwise, I support that you focus your efforts on making the software better and enriching the features.

**Reply by rodlie** [reply]
I will reconsider flatpak at some point, but if the package solution degrades the application experience, I will not support it.
